jujugui/static/gui/src/app/store/charmstore-api.js

```diff
@@ -53,33 +53,11 @@ YUI.add('charmstore-api', function(Y) {
         with a response of >= 400.
     */
     _makeRequest: function(path, successCallback, failureCallback) {
-      this.requestHandler.sendGetRequest(
-          path,
-          // The WebHandler methods allow you to pass in headers, username,
-          // password, progressCallback which we do not need.
-          null, null, null, null,
-          this._requestHandler.bind(this, successCallback, failureCallback));
-    },
-
-    /**
-      Handles the request response from the _makeRequest method, calling the
-      supplied failure callback if the response status was >= 400 or passing the
-      response object to the supplied success callback.
-
-      @method _requestHandler
-      @param {Function} successCallback Called when the api request completes
-        successfully.
-      @param {Function} failureCallback Called when the api request fails
-        with a response of >= 400.
-      @param {Object} response The XHR response object.
-    */
-    _requestHandler: function(successCallback, failureCallback, response) {
-      var target = response.target;
-      if (target.status >= 400) {
-        failureCallback(response);
-        return;
-      }
-      successCallback(response);
+      this.bakery.sendGetRequest(path,
+        this.charmstoreURL + this.apiPath + '/set-auth-cookie',
```

Comment:
I'm not saying to not have it be configurable, but instead of sending it for every request set it when you instantiate the bakery instance.